pkg/internal/controllers/certificaterequestpolicies.go

```diff
@@ -224,7 +226,7 @@ func (c *certificaterequestpolicies) reconcileStatusPatch(ctx context.Context, r
 			},
 		)
 
-		return result, policyPatch, nil
+		return result, &policy.Status, nil
```

Comment:
I think the logic must be more delicate than this. How do we achieve the removal of conditions that are no longer relevant? We user server-side apply (SSA) in this controller, and the idea is to set just the fields we care for. I agree we must consider existing conditions, but I still think we should prepare a status patch.

Reply:
I understand, it makes sense. I'll make changes.